stepper/stepper_controller/include/L6470_pkg/L6470_lib.py

```diff
@@ -0,0 +1,563 @@
+        return Status & 0x0000FFFF
+
+    # ==============================================================================================
+    # Stepper command shortcuts
+    # These functions are shortcuts for using the stepper commands with a specific register.
+    # ==============================================================================================
+    # CurrentPosition command
+    # Obtain the current absolute position of the stepper.
+    def currentPosition(self):
+        # Obtain position, in two's complement form.
+        Pos = self.getParam(self.REG_ABS_POS)
+        
+        # Convert to int value.
+        if Pos & (1 << 21):  # If sign bit is set:
+            Pos = Pos - (1 << 22)  # Compute negative value.
+                
+        return Pos
+
+    # MarkPosition command
+    # Obtain the current marked position of the controller.
+    # The mark position is a saved position to which the GoMark command can easily go.
+    def markPosition(self):
+        # Obtain position, in two's complement form.
+        Mark = self.getParam(self.REG_MARK)
+        
+        # Convert to int value.
+        if Mark & (1 << 21):  # If sign bit is set:
+            Mark = Mark - (1 << 22)  # Compute negative value.
+                
+        return Mark
+
+    # CurrentSpeed command
+    # Obtain the speed at which the stepper is currently moving, in step/s.
+    def currentSpeed(self):
+        # Obtain value in SPEED register in step/tick and convert to step/s.
+        Speed = self.speedToStepSec(self.getParam(self.REG_SPEED))
+        
+        return Speed
+
+    # CurrentAcceleration command
+    # Obtain the configured acceleration, in step/s^2.
+    # This acceleration value is used for soft starts of the stepper.
+    def currentAcceleration(self):
+        # Obtain value in ACC register in step/tick^2 and convert to step/s^2.
+        Acc = self.accToStepSec(self.getParam(self.REG_ACC))
+        
+        return Acc
+
+    # CurrentDeceleration command
+    # Obtain the configured deceleration, in step/s^2.
+    # This deceleration value is used for soft stops of the stepper.
+    def currentDeceleration(self):
+        # Obtain value in DEC register in step/tick^2 and convert to step/s^2.
+        Dec = self.accToStepSec(self.getParam(self.REG_DEC))
+        
+        return Dec
+
+    # SetMinSpeed command
+    # Set a new minimum speed for the stepper, in step/s.
+    # The Optimization parameter set the low-speed optimization flag. If set to True, low-speed
+    # optimization will be activated and the minimum speed value will be ignored by the controller.
+    def setMinSpeed(self, MinSpeed, Optimized):
+        if MinSpeed < 0:
+            return
+        
+        # Limit value to 12 bits.
+        if MinSpeed > 0x0FFF:
+            MinSpeed = 0x0FFF
+        
+        # Add the low speed optimization bit if specified.
+        if Optimized == True:
+            MinSpeed += 0x1000
```

Comment:
Your way is probably cleaner, but here's a neat python golf trick:
`MinSpeed += (0, 0x1000)[Optimized]`